main.go

```diff
@@ -6,52 +6,65 @@ import (
 	"path/filepath"
 	"flag"
 
-	"github.com/docker/docker/builder/dockerignore"
-	"github.com/docker/docker/pkg/fileutils"
+	"./dockerignore"
+	"./fileutils"
 )
 
+// Declare the array of type String
 var excludes []string
 
 func main() {
+	// Package flag implements command-line flag parsing.
+	// Parse() parses the command-line flags from os.Args[1:]. Must be called after all flags are defined and before flags are accessed by the program.
 	flag.Parse()
-	root := flag.Arg(0)
-	if len(root) == 0 {
-		fmt.Printf ("Pass in file path to examine.\n")
-		panic("Missing main argument")
-	}
+	// Check the number of command-line arguments.
+	// It should be 2: the .go file itslef and the path to inspect.
+	if len(os.Args) != 2 {
+		panic("Please provide a valid path to be inspected.")
+	}	
 
-	cwd, _ := os.Getwd()
-	fmt.Printf("Will apply .dockerignore on path %v ( %v )\n", root, cwd)
-	file, dockerignoreerr := os.Open(".dockerignore")
-	check(dockerignoreerr)
+	// Dynamically assign the passed argument for further inspection.
+	root := flag.Arg(0)
+	
+	// Print out the path on which .dockerignore will be applied.
+	fmt.Println("Will apply .dockerignore on path", root)
+	// Opens the .dockerignore file for reading.
+	file, err := os.Open(".dockerignore")
+	// Checks if there is an error.
+	check(err)
 
-	excludes, _ = dockerignore.ReadAll(file)
-	fmt.Printf("Patterns are %v\n",excludes)
+	// Read and assign the content which needs to be excluded.
+	excludes, err = dockerignore.ReadAll(file)
+	// Check if there is an error.	
+	check(err)
```

Comment:
With first read, because of the extraneous changes I missed the vital requested change to actually check the error i.e. removing `_` and replacing with `err`.